lesson16/main.js

```diff
@@ -0,0 +1,111 @@
+const li = document.createElement("li");
+const contetns = document.getElementsByClassName("tab-content");
+const tabs = document.getElementsByClassName("tab-item");
+
+const displayInfo = (info) => {
+  const li = document.createElement("li");
+  if (typeof info === "string") {
+    li.textContent = info;
+    ul.appendChild(li);
+    return console.log(info);
+  }
+  throw new Error("Please pass a string value");
+};
+
+const displayErrorInfo = (error) => {
+  console.log(error);
+  li.textContent = `Failed to get data from server`;
+  ul.appendChild(li);
+};
+
+const getData = async (api) => {
+  let response = null;
+  try {
+    response = await fetch(api);
+  } catch (error) {
+    displayErrorInfo(error);
+  }
+  if (response.ok) {
+    return response.json();
+  } else {
+    displayErrorInfo(response);
+  }
+};
+
+const checkResponse = async (callBackFunction) => {
+  let response = await getData(
+    "https://mocki.io/v1/40cbd0f7-2a64-4b95-a991-57ff982901a8"
+  );
+  if (response.data) {
+    callBackFunction(response.data);
+  } else {
+    displayInfo("no data available");
+  }
+};
+
+const createTabs = (items) => {
+  ul.classList = "tab-menu";
+  const fragment = new DocumentFragment();
+  for (let item of items) {
+    const li = document.createElement("li");
+    li.textContent = item.category;
+    li.classList = "tab-item";
+    fragment.appendChild(li);
+  }
+  ul.appendChild(fragment);
+  activateFirstTab();
+};
+
+const activateFirstTab = () => {
+  tabs[0].classList.add("active");
+};
+
+// this function will be modified to render articles, comments, an image and icons.
+const createArticles = (items) => {
+  const fragment = new DocumentFragment();
+  const articleContainerDiv = document.createElement("div");
+  articleContainerDiv.classList.add("tab-box");
+  for (let item of items) {
+    const article = document.createElement("div");
+    article.classList.add("tab-content");
+    article.textContent = item.category;
+    fragment.appendChild(article);
+  }
+  articleContainerDiv.appendChild(fragment);
+  ul.insertAdjacentElement("afterend", articleContainerDiv);
+  activateFirstArticleContent();
+};
+
+const activateFirstArticleContent = () => {
+  contetns[0].classList.add("show");
+};
+
+const removeClassList = (className, index) => {
+  document.getElementsByClassName(className)[index].classList.remove(className);
+};
+
+const displayTabsAndContents = async () => {
+  await checkResponse(createTabs);
+  await checkResponse(createArticles);
+
+  const listOfTabs = Array.from(tabs);
+  const listOfContetns = Array.from(contetns);
+
+  listOfTabs.forEach((tab) => {
+    tab.addEventListener("click", (e) => {
+      removeClassList("active", 0);
+      tab.classList.add("active");
+
+      const indexNumber = listOfTabs.findIndex((tabContent) =>
+        tabContent.classList.contains("active")
+      );
+
+      removeClassList("show", 0);
+      listOfContetns[indexNumber].classList.add("show");
+    });
+  });
+};
```

Comment:
IMO
I felt there was no need to make it a function about ``removeClassList`` as well. It doesn't seem like it would be used elsewhere. But maybe that is just my personal feeling.

```suggestion
// const removeClassList = (className, index) => {
//   document.getElementsByClassName(className)[index].classList.remove(className);
// };
const activeClass = document.getElementsByClassName("active");
const showClass = document.getElementsByClassName("show");

const displayTabsAndContents = async () => {
  await checkResponse(renderTabs);
  await checkResponse(renderArticlesContents);

  const listOfTabs = Array.from(tabs);
  const listOfContetns = Array.from(contetns);

  listOfTabs.forEach((tab) => {
    tab.addEventListener("click", (e) => {
      activeClass[0].classList.remove("active");
      tab.classList.add("active");

      const indexNumber = listOfTabs.findIndex((tabContent) =>
        tabContent.classList.contains("active")
      );

      showClass[0].classList.remove("show");
      listOfContetns[indexNumber].classList.add("show");
    });
  });
};
```

Reply:
I agree with Chihiro-san!

Reply:
Thank you for the comment both of you 
Suggested code makes it more readable
Please see the updated code on this commit 
59fb045fc215cf1bab0148169691a59cf80422a1